### PR TITLE
fix: allow removing members from an API

### PR DESCRIPTION
**Pull request by kamiiiel**
see: https://gravitee.atlassian.net/browse/GKO-7

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-jobsxzvfhq.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java [hunk 1]
```diff
@@ -462,10 +474,47 @@ private void createOrUpdateMembers(final ExecutionContext executionContext, ApiE
...
+                 * can be removed
+                 */
+                if (apiEntity.getOriginContext().isOriginKubernetes()) {
+                    // by removing the member, we will eventually end up with a clean list of
+                    // Members that can be removed from this API
+                    membersAlreadyPresent.remove(memberToImport);
+                }
+            }
+
+            if (apiEntity.getOriginContext().isOriginKubernetes()) {
```
Comment (a-cordier) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: Not sure but I think we should check the origin on the apiJSONNode. If we enter an update from a CRD export, I'm not sure at all the apiEntity will have a kubernetes origin

**Review by a-cordier — CHANGES_REQUESTED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java (hunk 1 above)
Comment (a-cordier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: I mean using

```java
apiJsonNode.isKubernetesOrigin()
```

Comment (kamiiiel) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: I'll change it but but we already save the origin in DB so it should have origin in it

Comment (kamiiiel) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java]: Changed it to `apiJsonNode.isKubernetesOrigin()`

**Review by a-cordier — APPROVED** [review]

**Pull request merged**